### ISSUE TITLE
Review of `trainer.py` and `training_config.toml`

### ISSUE THREAD

**Issue by AmateurAcademic** [AmateurAcademic/nlu-engine#1]
# Description
I am unsure if the training script contains issues or problems that lead to reduced performance of the trained model. It might be a good idea to have someone review the training script. 

## Questions
* Is the use of the `attention_mask`, `padding` at `max_length`, the `max_length`, and `gradient_accumulation_steps` good for these tasks?
* Is the learning rate ideal?
* Should we use `eval_loss` for `metric_for_best_model`?
* `compute_metrics`: I am not very happy that we are using exact matches. For intent, this makes sense, as they are labels. But for the entity task, it can lead to lower performance. However, if we would use edit distance, it could lead to higher scores than deserved, but perhaps there is some other metric we should use for the entity task? Maybe ROUGE or something similar?

# DoD (Definition of Done)
- [x] The above questions are answered
- [x] The training and config are adapted where needed
- [x] The changes are tested and benchmarked

**Comment by AmateurAcademic** [comment]
It seems like the learning rate was set too high. Strangely, I didn't have issues with it before. I will still implement ROUGE and see if that works better than the exact match. I do wonder if I can implement that for entity tagging and keep the exact match for intent matching.
